xcat-inventory/xcclient/inventory/invdiff

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/python
+from __future__ import print_function
+import deepdiff
+import pprint
+import yaml
+import json
+import sys
+import subprocess
+import re
+
+
+class InvalidFileException(Exception):
+    pass
+
+def runCommand(cmd, env=None):
+    """
+    Run one command only, when you don't want to bother setting up
+    the Popen stuff.
+        (retcode,out,err)=runCommand('lsxcatd -v')
+    """
+    try:
+        p = subprocess.Popen(cmd,
+        env=env,
+        shell=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE)
+        out, err = p.communicate()
+    except OSError,e:
+        return p.returncode,out, err
+    return p.returncode,out, err
+
+
+def loadfile(filename):
+    contents={}
+    fmt = 'json'
+    with open(filename,"r") as fh:
+        f=fh.read()
+        if not f.startswith('{'):
+            fmt = 'yaml'
+        try:
+            contents=json.loads(f)
+        except ValueError:
+            try:
+                contents = yaml.load(f)
+            except Exception,e:
+                raise InvalidFileException("Error: failed to load file \"%s\", please validate the file with 'yamllint %s'(for yaml format) or 'cat %s|python -mjson.tool'(for json format)!"%(filename,filename,filename))
+        return contents, fmt
+    return None, fmt
+
+
+class format_diff_output(object):
+
+    def __init__(self, format_type):
+        self.format_type = format_type 
+        self.flag_dict = {'-diff': '-', '+diff': '+'}
+
+    def _get_path_as_list(self, path):
+        path_str = path.replace(']', '')
```

Comment:
Please use `re.findall` to pick out the sub key list, refer to https://www.guru99.com/python-regular-expressions-complete-tutorial.html#7